tests/jballerina-unit-test/src/test/resources/test-src/statements/arrays/array_fill_runtime_test.bal

```diff
@@ -144,6 +144,71 @@ function tesTwoDimensionalArrayWithConstantSizeReferenceFill() {
     assertEqualPanic([1,2,3], b[1]);
 }
 
+function testAnonRecordTypeWithDefaultValueFill() {
+    record {| int i = 101; |}[3] testArray = [];
+    testArray[1] = {i: 202};
+    assertEqualPanic(3, testArray.length());
+    assertEqualPanic(101, testArray[0].i);
+    assertEqualPanic(202, testArray[1].i);
+    assertEqualPanic(101, testArray[2].i);
+
+    testAnonRecordAsFunctionReturnType();
+}
+
+function testAnonRecordAsFunctionReturnType() {
+    var resultArr = getAnonFromAnon();
+    resultArr[1] = {name: "Jack"};
+    assertEqualPanic(2, resultArr.length());
+    assertEqualPanic("Jane", resultArr[0].name);
+    assertEqualPanic("Jack", resultArr[1].name);
```

Comment:
We can move the common code to a separate function and invoke the function with different parameters